packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart

```diff
@@ -455,12 +455,6 @@ class AllocationProfileTableView extends StatefulWidget {
 
 class AllocationProfileTableViewState
     extends State<AllocationProfileTableView> {
-  @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
-    widget.controller.initialize();
```

Comment:
this is changing the lifecycle of when we pay the cost to initialize the `ProfilePaneController`. The way this code used to be delayed controller initialization until we actually needed it when the Allocation profile view was opened. Are we certain this is what we want to do?

Reply:
It is job of memory controller, not widget, to initiate initialization of ProfilePaneController.

Reply:
Performing the initialization in the widget lifecycle makes this lazy, which we probably still want. Otherwise we will pay the cost of listeners firing in the background for a feature that may not even be in use. This is similar to how we don't initialize the memory controller if we are on the flutter inspector, for example. We initialize on first use.